folium/features.py

```diff
@@ -241,8 +241,19 @@ def __init__(self, data, style_function=None):
             else:  # This is a filename
                 self.embed = False
                 self.data = data
+        elif data.__class__.__name__ in ['GeoDataFrame','GeoSeries']:
```

Comment:
I made my peace with this.  I did not like it at first but now I see it is best.

PS: Can you add a space after the comma?  To avoid pep8 issues later on.